File: mobile/app/src/main/java/com/example/hkotisk_mobile/adapter/OrderAgainAdapter.java
package com.example.hkotisk_mobile.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.hkotisk_mobile.R;
import com.example.hkotisk_mobile.model.OrderAgainItem;

import java.util.List;

public class OrderAgainAdapter extends RecyclerView.Adapter<OrderAgainAdapter.OrderAgainViewHolder> {
    private List<OrderAgainItem> orderAgainItems;

    public OrderAgainAdapter(List<OrderAgainItem> orderAgainItems) {
        this.orderAgainItems = orderAgainItems;
    }

    @NonNull
    @Override
    public OrderAgainViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_order_again, parent, false);
        return new OrderAgainViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull OrderAgainViewHolder holder, int position) {
        OrderAgainItem item = orderAgainItems.get(position);
        holder.imageView.setImageResource(item.getImageResourceId());
        holder.nameTextView.setText(item.getName());
        holder.availabilityTextView.setText("Available: " + item.getAvailable());
        holder.soldTextView.setText("Sold: " + item.getSold());
        holder.priceTextView.setText(String.format("₱%.2f", item.getPrice()));
    }

    @Override
    public int getItemCount() {
        return orderAgainItems.size();
    }

    static class OrderAgainViewHolder extends RecyclerView.ViewHolder {
        ImageView imageView;
        TextView nameTextView;
        TextView availabilityTextView;
        TextView soldTextView;
        TextView priceTextView;

        OrderAgainViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.order_again_image);
            nameTextView = itemView.findViewById(R.id.order_again_name);
            availabilityTextView = itemView.findViewById(R.id.order_again_availability);
            soldTextView = itemView.findViewById(R.id.order_again_sold);
            priceTextView = itemView.findViewById(R.id.order_again_price);
        }
    }
} 

File: mobile/app/src/main/java/com/edu/cit/hkotisk/adapter/OrderAgainAdapter.kt
package com.edu.cit.hkotisk.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.edu.cit.hkotisk.databinding.ItemOrderAgainBinding
import com.edu.cit.hkotisk.model.OrderAgainItem

class OrderAgainAdapter(private val orderAgainItems: List<OrderAgainItem>) : 
    RecyclerView.Adapter<OrderAgainAdapter.OrderAgainViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OrderAgainViewHolder {
        val binding = ItemOrderAgainBinding.inflate(
            LayoutInflater.from(parent.context), 
            parent, 
            false
        )
        return OrderAgainViewHolder(binding)
    }

    override fun onBindViewHolder(holder: OrderAgainViewHolder, position: Int) {
        val item = orderAgainItems[position]
        holder.bind(item)
    }

    override fun getItemCount(): Int = orderAgainItems.size

    class OrderAgainViewHolder(private val binding: ItemOrderAgainBinding) : 
        RecyclerView.ViewHolder(binding.root) {
        
        fun bind(item: OrderAgainItem) {
            binding.orderAgainImage.setImageResource(item.imageResourceId)
            binding.orderAgainName.text = item.name
            binding.orderAgainAvailability.text = "Available: ${item.available}"
            binding.orderAgainSold.text = "Sold: ${item.sold}"
            binding.orderAgainPrice.text = String.format("₱%.2f", item.price)
        }
    }
} 